015/master

BenGitsCode opened the pull request:
@gaand can you review this? Still testing but I've run through the api-guide exercises with success, and want to get this to jason asap.

BenGitsCode committed:
Update from rails-api-template

Refresh for cohort 015

BenGitsCode committed:
Change name to library-api-guide

- Change all instances of library-api to library-api-guide

- Address and close #11

BenGitsCode committed:
Change host from 3000 to 4741

- Change all instances of host `3000` to `4741`

- Also updates `pg` to gem files

BenGitsCode committed:
Add rake:db reset instructions

- to avoid nuke_pave
- commands to reset without dropping db
- refactored to oneline to mitigate errors in intermediate commands.
- Address and close #7

BenGitsCode committed:
Clarify two separate secrets in README

- Clarify language slightly to indicate two different secrets.

- Feels like a silly clarification, but I've seen this lead to
confusion in previous cohorts.

gaand commented on Gemfile:
```diff
@@ -5,6 +5,7 @@ gem 'rails', '~> 4.2.6'
 gem 'rails-api'
 gem 'active_model_serializers', '~> 0.10.0'
 gem 'rack-cors', require: 'rack/cors'
+gem 'rb-readline'
```
I don't believe this is correct.  The latest template does not include rb-readline.

gaand replied:
I don't believe this is correct. The latest template does not include rb-readline.

BenGitsCode committed:
Remove rb-readline from gems

- rb-readline not used in latest update of rails-api-template

- Addressed in Pull request comment #12

BenGitsCode commented:
`rb-readline` is not in rails-api-template but when I remove it and bundle install, I get the following error on `bundle exec rake secret`

``` sh
 Sorry, you can't use byebug without Readline. To solve this, you need to
    rebuild Ruby with Readline support. If using Ubuntu, try `sudo apt-get
    install libreadline-dev` and then reinstall your Ruby.
rake aborted!
LoadError: dlopen(/usr/local/var/rbenv/versions/2.3.1/lib/ruby/2.3.0/x86_64-darwin15/readline.bundle, 9): Library not loaded: /usr/local/opt/readline/lib/libreadline.6.dylib
  Referenced from: /usr/local/var/rbenv/versions/2.3.1/lib/ruby/2.3.0/x86_64-darwin15/readline.bundle
  Reason: image not found - /usr/local/var/rbenv/versions/2.3.1/lib/ruby/2.3.0/x86_64-darwin15/readline.bundle
/Users/Benjamski/wdi-000/tmp/rails-api-template/config/application.rb:16:in `<top (required)>'
/Users/Benjamski/wdi-000/tmp/rails-api-template/Rakefile:4:in `require'
/Users/Benjamski/wdi-000/tmp/rails-api-template/Rakefile:4:in `<top (required)>'
/usr/local/var/rbenv/versions/2.3.1/bin/bundle:23:in `load'
/usr/local/var/rbenv/versions/2.3.1/bin/bundle:23:in `<main>'
(See full trace by running task with --trace)
```

So yes, the rails-api-template does not have `rb-readline` but I just tested it in rails-api-template and get the same error, so the question is should it be there? Or am I missing something?

BenGitsCode commented:
When I install `rb-readline` locally everything runs swimmingly, but it's a hot mess without it and I always get the above error. @gaand tested this pull on his machine with no issue so think the problem is specific to my machine. Working through that as time permits but want to make sure @J-Weeks has this for his training repos.

gaand commented on README.md:
```diff
@@ -10,8 +10,8 @@ A simple API for books.
-1.  Generate new `development` and `test` secrets (`bundle exec rake secret`).
-1.  Store them in `.env` with keys `SECRET_KEY_BASE_<DEVELOPMENT|TEST>`
+1.  Generate two new secrets, `development` and `test` (`bundle exec rake secret`).
+1.  Store them in `.env` with keys `SECRET_KEY_BASE_<DEVELOPMENT>` and `SECRET_KEY_BASE_<TEST>`
```
This change seems unnecessary.  Why is it here?

BenGitsCode replied:
Happy to revert it, I only changed it because I remember in wdi-11 there being confusion on needing to generate two separate keys. I'll remove and re-push.

BenGitsCode committed:
Revert "Clarify two separate secrets in README"

This reverts commit 3831cc3ac61f02ff55966efb1247ba8ba1405ff1.

Removing changed based on comment in pull request #12
